engine/Assets/Scripts/Physics/CustomWheel.cs

```diff
@@ -63,61 +61,63 @@ public class CustomWheel : MonoBehaviour {
     /// Calculates friction forces for the wheel and applies them. Should be run every FixedUpdate
     /// </summary>
     /// <param name="mod">Modifier to account for janky Unity joints</param>
-    public void GetFrictionForces(float mod) {
-        if (_pairings.Count > 0) {
-            CalculateFriction();
-
-            _lastImpulseTotal = (_staticImpulseVecAccum + _rollingImpulseVecAccum);
-
-            Rb.velocity += _lastImpulseTotal * mod;// / Rb.mass;
+    public void CalculateAndApplyFriction(float mod)
+    {
+        if (!HasContacts)
+            return;
 
-            _staticImpulseVecAccum = new Vector3();
-            _rollingImpulseVecAccum = new Vector3();
-        }
+        _lastImpulseTotal = CalculateNetFriction();
+        
+        Rb.velocity += _lastImpulseTotal * mod;// / Rb.mass;
     }
-
-    // Tbh these variables are relics but I don't want to mess with anything
-    private Vector3 _staticImpulseVecAccum = new Vector3();
-    private Vector3 _rollingImpulseVecAccum = new Vector3();
-    public void OnCollisionStay(Collision collision) {
-        // _collisionCalls++;
-        _pairings.Add((collision.impulse, collision.relativeVelocity));
+    
+    public void OnCollisionStay(Collision collision)
+    {
+        // Flip impulse vector if for whatever reason it is backwards (happens with mean machine robot)
+        Vector3 impulse = (collision.impulse.y < 0f) ? -collision.impulse: collision.impulse;
```

Comment:
Change to recalculated impulse vector using point of contact and center of mass as discussed.

Reply:
Done